google_guest_agent/snapshot_listener.go

```diff
@@ -0,0 +1,196 @@
+//  Copyright 2020 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	pb "github.com/GoogleCloudPlatform/guest-agent/google_guest_agent/protos"
+	"github.com/GoogleCloudPlatform/guest-logging-go/logger"
+	"github.com/hashicorp/golang-lru"
+	"google.golang.org/grpc"
+)
+
+var (
+	seenPreSnapshotOperationIds, _  = lru.New(128)
+	seenPostSnapshotOperationIds, _ = lru.New(128)
+)
+
+type snapshotConfig struct {
+	Timeout               int // seconds
```

Comment:
nit: these should be private fields as the struct is private (unless there is some need like the json marshaller)